File: practices/ListViewsSpinners/app/src/main/java/mx/itesm/listviewsspinners/MainActivity.java
package mx.itesm.listviewsspinners;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Spinner;
import android.widget.Toast;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity implements
        AdapterView.OnItemClickListener,
        AdapterView.OnItemSelectedListener {

    private ListView listView;
    private Spinner spinner;

    private ArrayList<Student> source2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        listView = (ListView)findViewById(R.id.listView);
        spinner = (Spinner)findViewById(R.id.spinner);

        // como poblar widgets para colecciones
        // datos - Adapter - widget

        String[] source = {"Hermes", "Alejandro", "Edy", "Sara"};

        // Adapter
        ArrayAdapter<String> adapter = new ArrayAdapter<String>(
                this,
                android.R.layout.simple_list_item_1,
                source
        );

        source2 = new ArrayList<>();

        source2.add(new Student("Marco", 60));
        source2.add(new Student("Marco", 60));
        source2.add(new Student("Marco", 60));
        source2.add(new Student("Marco", 60));
        source2.add(new Student("Marco", 60));
        source2.add(new Student("Marco", 60));
        source2.add(new Student("Marco", 60));
        source2.add(new Student("Marco", 60));
        source2.add(new Student("Miguelito", 30));
        source2.add(new Student("Ponchito", -20));
        source2.add(new Student("Katia", 68));

        StudentAdapter adapter2 = new StudentAdapter(source2, this);

        listView.setAdapter(adapter2);
        spinner.setAdapter(adapter2);

        listView.setOnItemClickListener(this);
        spinner.setOnItemSelectedListener(this);
    }


    @Override
    public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
        Student selected = source2.get(i);
        Toast.makeText(this, selected.getNombre(), Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
        Student selected = source2.get(i);
        Toast.makeText(this, selected.getNombre(), Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onNothingSelected(AdapterView<?> adapterView) {

    }
}

File: homeworks/activity3.1/app/src/main/java/mx/itesm/activity31/MainActivity.kt
package mx.itesm.activity31

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.AdapterView
import kotlinx.android.synthetic.main.activity_main.*
import org.json.JSONException
import org.json.JSONArray
import org.json.JSONObject
import android.widget.Toast





class MainActivity : AppCompatActivity(), AdapterView.OnItemClickListener{

    var source: ArrayList<Food> = ArrayList()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val jsonTest = "[{'name':'burger', 'price':15, 'description':'a juicy burger!'}," +
                "{'name':'hotdog', 'price':20, 'description':'an ok hot dog'}," +
                "{'name':'tacos', 'price':12, 'description':'some pretty good tacos'}," +
                "{'name':'torta', 'price':22, 'description':'nice torta'}," +
                "{'name':'carne asada', 'price':50, 'description':'a great carne asada'}]"

        try {
            val obj = JSONArray(jsonTest)

            for (i in 0 until obj.length()) {
                val actual = obj.get(i) as JSONObject
                source.add(Food(actual.getString("name"), actual.getString("price"),actual.getString("description")))
            }

        } catch (e: JSONException) {
            e.printStackTrace()
        }

        val adapter = FoodAdapter(this, source)

        listView.adapter = adapter
        listView.onItemClickListener = this

    }

    override fun onItemClick(adapterView: AdapterView<*>?, view: View?, i: Int, l: Long) {
        val selected = source.get(i)
        Toast.makeText(this, selected.description, Toast.LENGTH_SHORT).show()
    }
}
